executor/cook/executor.py

```diff
@@ -218,7 +222,46 @@ def get_task_state(exit_code):
         return cook.TASK_FINISHED
 
 
-def manage_task(driver, task, stop_signal, completed_signal, config, stdout_name, stderr_name):
+def set_environment(environment, key, value):
+    """Updates an entry in the environment dictionary.
+
+    Returns
+    -------
+    Nothing.
+    """
+    if key not in environment or environment[key] != value:
+        logging.info('Setting process environment[{}]={}'.format(key, value))
+        environment[key] = value
+
+
+def retrieve_process_environment(config, os_environ):
+    """Prepares the environment for the subprocess.
+    The function also ensures that env[EXECUTOR_PROGRESS_OUTPUT_FILE_ENV]
+    and PROGRESS_OUTPUT_FILE is set to config.progress_output_name in the
+    returned environment.
+
+    Parameters
+    ----------
+    config: cook.config.ExecutorConfig
+        The current executor config.
+    os_environ: dictionary
+        A dictionary representing the current environment.
+
+    Returns
+    -------
+    The environment dictionary for the subprocess.
+    """
+    environment = dict(os_environ)
+
+    progress_output_name = config.progress_output_name
+    custom_progress_output_file_env = environment.get('EXECUTOR_PROGRESS_OUTPUT_FILE_ENV', '')
```

Comment:
We're already checking for `EXECUTOR_PROGRESS_OUTPUT_FILE_ENV` in `config.py`.  The config could know the name of the env var to set with the progress output file, whether it's customized or not.  There need not be branching in this function.  It's responsibility can be to simply set the value of  configured environment variable to the correct value for the progress output file.

Reply:
Do we set an environment variable to the progress output file if `EXECUTOR_PROGRESS_OUTPUT_FILE_ENV` isn't set?

Reply:
> Do we set an environment variable to the progress output file if EXECUTOR_PROGRESS_OUTPUT_FILE_ENV isn't set?

No. That is what I was doing with `PROGRESS_OUTPUT_FILE` previously.

Reply:
> The config could know the name of the env var to set with the progress output file, whether it's customized or not. There need not be branching in this function. It's responsibility can be to simply set the value of configured environment variable to the correct value for the progress output file.

Even if config knows the name, the branching would then be if the name is a non-empty string before setting the configured environment. Is your goal to limit the exposure of the `EXECUTOR_PROGRESS_OUTPUT_FILE_ENV` string?

Reply:
@sradack any thoughts on this?